ServerSide/yoty/Core/Managers/Bids/BidsManager.cs

```diff
@@ -0,0 +1,291 @@
+﻿// Copyright (c) YOTY Corporation and contributors. All rights reserved.
+
+namespace YOTY.Service.Core.Managers.Bids
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Runtime.CompilerServices;
+    using System.Threading.Tasks;
+    using YOTY.Service.Data;
+    using YOTY.Service.Data.Entities;
+    using YOTY.Service.WebApi.PublicDataSchemas;
+
+    public class BidsManager : IBidsManager
+    {
+        private const string BidNotFoundFailString = "Failed, Bid not found";
+        private static YotyContext _context = new YotyContext();
+
+        public async Task<Response<BuyerDTO>> AddBuyer(BidBuyerJoinRequest bidBuyerJoinRequest)
+        {
+            BidEntity bid = await _context.Bids.FindAsync(bidBuyerJoinRequest.bidId).ConfigureAwait(false);
+            if (bid == null)
+            {
+                return new Response<BuyerDTO>() { DTOObject = null, IsOperationSucceeded = false, SuccessOrFailureMessage = BidNotFoundFailString };
+            }
+            bid.CurrentParticipancies.Add(new ParticipancyEntity {
+                BidId = bidBuyerJoinRequest.bidId,
+                BuyerId = bidBuyerJoinRequest.buyerId,
+                NumOfUnits = bidBuyerJoinRequest.Items
+            });
+            bid.UnitsCounter += bidBuyerJoinRequest.Items;
+            using (var new_context = new YotyContext())
+            {
+                try
+                {
+                    new_context.Bids.Update(bid);
+                    await new_context.SaveChangesAsync().ConfigureAwait(false);
+                }
+                catch (Exception ex)
+                {
+                    return new Response<BuyerDTO>() { DTOObject = null, IsOperationSucceeded = false, SuccessOrFailureMessage = ex.Message };
+                }
+            }
+            //TODO dropping the dto obj from the response will save us the second db access 
+            var buyer_ent = await _context.Buyers.FindAsync(bidBuyerJoinRequest.buyerId).ConfigureAwait(false);
+            //TODO map ent to dto
+            return new Response<BuyerDTO>() { DTOObject = null, IsOperationSucceeded = true, SuccessOrFailureMessage = this.getSuccessMessage() };
+        }
+
+        public async Task<Response<SupplierProposalDTO>> AddSupplierProposal(SupplierProposalRequest supplierProposal)
+        {
+            BidEntity bid = await _context.Bids.FindAsync(supplierProposal.BidId).ConfigureAwait(false);
+            if (bid == null)
+            {
+                return new Response<SupplierProposalDTO>() { DTOObject = null, IsOperationSucceeded = false, SuccessOrFailureMessage = BidNotFoundFailString };
+            }
+            bid.CurrentOffers.Add(new SellerOfferEntity {
```

Comment:
we should have same names between all kind of schemas of a object ... 
replace Seller with Supplier  and Offered with Proposed